hugo/lib/createArticle.js

```diff
@@ -0,0 +1,56 @@
+const moment = require('moment');
+const marked = require('marked');
+const fs = require('fs');
+
+module.exports = (entry) => {
+  let content = entry.fields;
+
+  // Grab Article information used build template meta data
+  let {
+    title,
+    body,
+    date,
+    category,
+    tags,
+    slug,
+    description
+  } = content;
+  let type = 'article';
+  let cleanTitle = title.replace(/\"/g, '\\"');
+  let cleanDescription = description.replace(/\"/g, '\\"');
+  let cleanBody = marked(body);
+  let headerPhotoInfo = content.headerPhoto.fields;
+
+  // Grab Author information
+  let {bio, name} = content.author.fields;
+  let authorImageInfo = content.author.fields.picture.fields;
+
+  // Use title for the filename. Lowercase, replace ' ' with '-', and only keep alphanumeric characters
+  let filename;
+  if (slug) {
+    filename = slug;
+  } else {
+    filename = title.toLowerCase().replace(/\ /g, '-').replace(/[^0-9a-zA-Z_-]/g, '');
+  }
+  // create markdown. There has to be a way to do this better
```

Comment:
I wouldn't feel too bad about this. I can look at the code and see that it's prepending an additional section to the body content before writing it.